src/Drivers/nlpPriDec_ex9_driver.cpp

```diff
@@ -0,0 +1,188 @@
+#include "nlpPriDec_ex9.hpp"
+//the solver
+#include "hiopAlgPrimalDecomp.hpp"
+
+#ifdef HIOP_USE_MAGMA
+#include "magma_v2.h"
+#endif
```

Comment:
Perhaps this should check if GPU is used?

Reply:
I think magma variable is set based on HIOP_USE_CUDA/GPU, so iiuc this should be good

Reply:
+1 to this being ok as it is clearer in context. Having a CMake variable for each TPL means the logic for them can always be taken care of in a `CMakeLists.txt` somewhere else, although functionally speaking I think it would be equivalent to just use `HIOP_USE_GPU` here.

Reply:
This has been changed as suggested.